### PR TITLE
🚀 5.0.0

### LINKED ISSUE #212: Task: Drop Python 3.7 (EOL)
Its support should be dropped.

https://devguide.python.org/versions/

### LINKED ISSUE #255: Buttons not showing on Raspberry Pi 5
Hi,

did someone encounter this issue: octorelay buttons not showing in the navbar?
Im using octoprint 1.9.3 Python 3.11.2 on a Raspberry Pi 5 8Gb Board with default plugins installed

TNX

### PR DESCRIPTION
**Pull request by RobinTail**
Closes #255 
Closes #212 

- #274 
- #281 
- #215 
- #282 
- #283

### CODE REVIEW

**Comment by RobinTail** [comment]
I'm getting the following error in logs when running on Raspberry Pi 4 (Octopi):

```
    import _lgpio
ImportError: /lib/arm-linux-gnueabihf/libc.so.6: version `GLIBC_2.33' not found (required by /home/pi/oprint/lib/python3.9/site-packages/_lgpio.cpython-39-arm-linux-gnueabihf.so)
```

**Comment by ch3p4ll3** [comment]
How did you installed octoprint and the plugin? I tried the lib also on my pi3b+ creating a new venv and works without any problem

**Comment by RobinTail** [comment]
Octopi 1.0 is Debian "bullseye" RpiOS 2022-09-22, @ch3p4ll3 
https://github.com/guysoft/OctoPi/releases/tag/1.0.0

**Comment by ch3p4ll3** [comment]
Could you try manually install the [library](https://github.com/joan2937/lg)?

**Comment by ch3p4ll3** [comment]
I'll also try also in my 3B+ with octopi

**Comment by RobinTail** [comment]
> Could you try manually install the [library](https://github.com/joan2937/lg)?

I installed it, @ch3p4ll3 , but it I think it should also somehow be connected to python (instead of regular `pip install lgpio`)

```
install -m 0644 lgpio.h                  /usr/local/include
install -m 0644 rgpio.h                  /usr/local/include
install -m 0755 liblgpio.so.1 /usr/local/lib
install -m 0755 librgpio.so.1 /usr/local/lib
install -m 0755 rgpiod                   /usr/local/bin
install -m 0755 rgs                      /usr/local/bin
install -m 0644 rgpiod.1                 /usr/local/man/man1
install -m 0644 rgs.1                    /usr/local/man/man1
install -m 0644 lgpio.3                  /usr/local/man/man3
install -m 0644 rgpio.3                  /usr/local/man/man3
ldconfig
/usr/bin/python3
/usr/bin/python3
```

**Comment by ch3p4ll3** [comment]
I think I found the problem. the package `libc6` in the repo is too old. Current version is: [2.38](https://tracker.debian.org/pkg/glibc)

**Comment by RobinTail** [comment]
Exactly, @ch3p4ll3 , my findings were similar.
Related issue: https://github.com/guysoft/OctoPi/issues/823

**Comment by RobinTail** [comment]
Anyway, does it work on Raspberry 5 with another OS, @ch3p4ll3 ?
Do you see the buttons? Do they operate relays?

**Comment by RobinTail** [comment]
That's a great idea, @ch3p4ll3 . Yes, Bullseye has `ldd` version 2.31. If you could make it select the proper factory somehow, that would make it compatible with current OctoPi.

**Comment by RobinTail** [comment]
Btw, I read that gpiozero can decide on the factory itself, somehow, if you don't define it explicitly. Maybe we can install both and it will choose the right one?

**Comment by ch3p4ll3** [comment]
> Btw, I read that gpiozero can decide on the factory itself, somehow, if you don't define it explicitly. Maybe we can install both and it will choose the right one? Sounds a bit optimistic probably.

Yeah we can try, Idk if it will also work with Pi5, I think it'll depend on the "loading" order of the pins factories

**Comment by ch3p4ll3** [comment]
Ok, just tested. the first pinFactory that gpiozero tries is `lgpio`, so I think this will work. Tomorrow I'll make some commits and test in my Pi5

**Comment by RobinTail** [comment]
I found the list of default factories used when none is specified:
https://github.com/gpiozero/gpiozero/blob/f3a338e56c28dd2d904016b3a8594f82c47da291/gpiozero/devices.py#L273-L286

**Comment by RobinTail** [comment]
🚀 5.0.0a5 published, @ch3p4ll3 
Please check that is still works Raspberry 5

**Comment by ch3p4ll3** [comment]
Yep, still working on my Pi5

**Comment by RobinTail** [comment]
✅ Regression test on Raspberry Pi 4 with OctoPi 1.0.0 (Bullseye) passed!
Thank you @ch3p4ll3

**Comment by RobinTail** [comment]
> Yep, still working on my Pi5

Perfect! I think it's a great achievement. I highly appreciate your effort and I'm glad with our collaboration.

I'm going to perform some chores and make a couple more tests before the final release.
Thank you so much, @ch3p4ll3

File: octoprint_octorelay/__init__.py [hunk 1]
```diff
@@ -108,10 +108,11 @@ def handle_list_all_command(self) -> Listing:
         settings = self._settings.get([], merged=True) # expensive
         for index in RELAY_INDEXES:
             if bool(settings[index]["active"]):
-                relay = Relay(
+                relay = Relay.get_or_create_relay(
                     int(settings[index]["relay_pin"] or 0),
                     bool(settings[index]["inverted_output"])
                 )
+
```
Comment (RobinTail) [inline octoprint_octorelay/__init__.py:115]:
```suggestion
```

**Comment by RobinTail** [comment]
🚀 5.0.0rc1
✅ QA passed

**Comment by RobinTail** [comment]
🚀 5.0.0rc2
@ch3p4ll3 , please confirm one last time that the version `5.0.0rc2` still works on Raspberry Pi 5

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,18 @@
 # Changelog
 
+## Version 5
+
+### 5.0.0
+
+- **Breaking changes**:
+  - Minimum version of Python supported: 3.9;
+- Feature: Supporting Raspberry Pi 5:
+  - Using `gpiozero` with two `lgpio` and `RPi.GPIO` factories depending on your hardware;
```
Comment (RobinTail) [inline CHANGELOG.md:10]:
```suggestion
  - Using `gpiozero` with `lgpio` or `RPi.GPIO` depending on your hardware;
```

**Review by RobinTail — COMMENTED** [review]
ready

**Comment by RobinTail** [comment]
Try to refresh the page hardly (Cmd+Shift+R) or re-login, @ch3p4ll3 
CSRF means that the session key of the UI does not match the running API (probably after the update)

**Comment by RobinTail** [comment]
Marketplace listing update: https://github.com/OctoPrint/plugins.octoprint.org/pull/1311

**Pull request merged**